lib/src/util/vulnerabilitySeverityLevels.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+/**
+ * All Possible Vulnerability (CVSS) Severity Levels.
+ */
+export const allVulnerabilitySeverityLevels = [
+  'none',
+  'low',
+  'medium',
+  'high',
+  'critical'
+] as const;
+
+type SingleDigit = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9';
+
+export type VulnerabilityScore = `${SingleDigit}.${SingleDigit}` | '10.0';
+
+/**
+ * The type containing only the possible vulnerability CVSS severity level values.
+ */
+export type VulnerabilitySeverityLevel =
+  (typeof allVulnerabilitySeverityLevels)[number];
+
+/**
+ * Lists the ranges of Vulnerability (CVSS) scores and their corresponding severity level.
+ */
+export const vulnerabilitySeverityAndScoreRangeList = Object.freeze([
+  { min: 9.0, max: 10.0, severity: 'critical' },
+  { min: 7.0, max: 8.9, severity: 'high' },
+  { min: 4.0, max: 6.9, severity: 'medium' },
+  { min: 0.1, max: 3.9, severity: 'low' },
+  { min: 0.0, max: 0.0, severity: 'none' }
+] as const);
+
+/**
+ * Maps vulnerability CVSS severity levels to their corresponding score ranges string representation.
+ */
+export const vulnerabilitySeverityScoreRangeMap = Object.freeze({
+  critical: '9.0\u201410.0',
```

Comment:
This structure can be derived from `vulnerabilitySeveriyAndScoreRangeList` rather than hardcoded.  At the same time though, this structures purpose seems very presentation-oriented, and potentially only relevant to one or two areas.  Does it really make sense as a reusable utility at all?

Reply:
> This structure can be derived from `vulnerabilitySeverityAndScoreRangeList` rather than hardcoded. At the same time though, this structures purpose seems very presentation-oriented, and potentially only relevant to one or two areas. Does it really make sense as a reusable utility at all?

I was torn between this too. Currently, it's being used to correctly display ranges in tooltips in some of the vulnerability components.
And potentially it might be useful for labelling in downstream UIs down the road to enforce the correct range formatting.
But yeah it might not be useful enough to warrant it as an exported reusable utility

Reply:
> I was torn between this too. Currently, it's being used to correctly display ranges in tooltips in some of the vulnerability components.
And potentially it might be useful for labelling in downstream UIs down the road to enforce the correct range formatting.
But yeah it might not be useful enough to warrant it as an exported reusable utility

If it's used in multiple places within RSC my recommendation would be to keep it in a shared utility but not export it from `index.ts`.

Reply:
> > I was torn between this too. Currently, it's being used to correctly display ranges in tooltips in some of the vulnerability components.
> > And potentially it might be useful for labelling in downstream UIs down the road to enforce the correct range formatting.
> > But yeah it might not be useful enough to warrant it as an exported reusable utility
> 
> If it's used in multiple places within RSC my recommendation would be to keep it in a shared utility but not export it from `index.ts`.

I've excluded it from export here: [b10ed3e](https://github.com/sonatype/sonatype-react-shared-components/pull/1202/commits/b10ed3efc1bf44bc6beebc68d3aee91f55b9c8c9)